sound/soc/sof/loader.c

```diff
@@ -184,13 +193,19 @@ static int load_modules(struct snd_sof_dev *sdev, const struct firmware *fw)
 
 	/* parse each module */
 	module = (void *)fw->data + sizeof(*header);
+	remaining = fw->size;
```

Comment:
Is this correct?
fw->size equals header->file_size + sizeof(*header)), and you skip the header by doing
module = (void *)fw->data + sizeof(*header);

so,  should it be
remaining = fw->size - sizeof(*header?

Reply:
Thanks @plbossart . Yes, it should be remaining = fw->size - sizeof(*header). I also did a slight modification for readability.